File: SDS.Mobile/app/src/main/java/com/example/spawneddeliveryservice/models/DetailsTransportPackageDataModel.java
package com.example.spawneddeliveryservice.models;

import org.json.JSONObject;

import java.util.Date;

public class DetailsTransportPackageDataModel {
    public int id;
    public double price;
    public int space;
    public int kilograms;
    public Date deadline;
    public String ownerName;
    public String ownerPhone;
    public String additionalContact;

    public static DetailsTransportPackageDataModel FromModel(String json){
        try {
            JSONObject jObject = new JSONObject(json);
            DetailsTransportPackageDataModel returnedModel = new DetailsTransportPackageDataModel();

            returnedModel.id = jObject.getInt("Id");
            returnedModel.price = jObject.getDouble("Price");
            returnedModel.space = jObject.getInt("Space");
            returnedModel.kilograms = jObject.getInt("Kilograms");
            String dateAsString = jObject.getString("Deadline");
            try {
                returnedModel.deadline = ConstantsDataModels.getDateFormatter().parse(dateAsString);
            }
            catch (Exception e) {
                returnedModel.deadline = ConstantsDataModels.getNoMsDateFormatter().parse(dateAsString);
            }
            returnedModel.ownerName = jObject.getString("OwnerName");
            returnedModel.ownerPhone = jObject.getString("OwnerPhone");
            returnedModel.additionalContact = jObject.getString("AdditionalContact");

            return returnedModel;
        }
        catch (Exception e){
            return null;
        }
    }
}

File: SDS.Clould/SDS.Clould.WebAPI/DataModels/DetailsTransportPackageDataModel.cs
namespace SDS.Clould.WebAPI.DataModels
{
    using System;
    using System.Linq;
    using System.Linq.Expressions;

    using SDS.Clould.Models;

    public class DetailsTransportPackageDataModel
    {
        public static Expression<Func<Package, DetailsTransportPackageDataModel>> FromData
        {
            get
            {
                return p => new DetailsTransportPackageDataModel
                {
                    Id = p.Id,
                    Price = p.Price,
                    Space = p.Space,
                    Kilograms = p.Kilograms,
                    Deadline = p.Deadline,
                    OwnerName = p.Owner.UserName,
                    OwnerPhone = p.Owner.PhoneNumber,
                    AdditionalContact = p.AdditionalContact
                };
            }
        }

        public int Id { get; set; }

        public decimal Price { get; set; }

        public int Space { get; set; }

        public int Kilograms { get; set; }

        public DateTime Deadline { get; set; }

        public string OwnerName { get; set; }

        public string OwnerPhone { get; set; }

        public string AdditionalContact { get; set; }
    }
}